qiskit_experiments/library/characterization/analysis/ramsey_xy_analysis.py

```diff
@@ -412,16 +424,93 @@ def _default_options(cls):
                 curve.ParameterRepr("f_err", "stark_ferr", "Hz"),
             ],
             data_subfit_map={
-                "Xpos": {"series": "X", "direction": "pos"},
-                "Ypos": {"series": "Y", "direction": "pos"},
-                "Xneg": {"series": "X", "direction": "neg"},
-                "Yneg": {"series": "Y", "direction": "neg"},
+                "Fpos": {"direction": "pos"},
+                "Fneg": {"direction": "neg"},
             },
             plotter=ramsey_plotter,
         )
 
         return options
 
+    def _format_data(
+        self,
+        curve_data: curve.ScatterTable,
+    ) -> curve.ScatterTable:
+        columns = list(curve_data.columns)
+
+        i = 0
+        # Reassign model class.
+        # These data are actually P1 value of Ramsey X, Y quadrature,
+        # but the data is automatically classified based on the fit models,
+        # which are for the synthetic phase data in this analysis.
+        # This yields false classification.
+        for series in ("X", "Y"):
+            for direction in ("pos", "neg"):
```

Comment:
This might be a little cleaner using `itertools.product` or `keys = ((s, d) for s in ("X", "Y") for d in ("pos", "neg"))` and then a single loop with `enumerate` like `for model_id, (series, direction) in enumerate(keys):`.

Reply:
This code is gone after the refactoring.